### PR TITLE
refactor: Move remaining members of container-utils to telemetry-utils

### PR DESCRIPTION
**Pull request by Josmithr**
Requires adding a dependency from `telemetry-utils` on `protocol-definitions`.

Moved API members:
* DataProcessingError 
* extractSafePropertiesFromMessage

The changes also attempt to better encapsulate / document parameter type information for the 2 moved members by adding a type alias.

Also...
* All lingering references to and dependencies on container-utils have been removed
* The package README has been updated to reflect its deprecated status
* Since the package no longer has tests, existing test infra has been removed

### CODE REVIEW

File: packages/utils/telemetry-utils/src/error.ts [hunk 1]
```diff
@@ -53,3 +60,118 @@ export class DataCorruptionError extends LoggingError implements IErrorBase, IFl
 		super(message, { ...props, dataProcessingError: 1 });
 	}
 }
+
+/**
+ * An incoming message from the Fluid Service.
+ */
+export type MessageLike = Partial<
```
Comment (Josmithr) [inline packages/utils/telemetry-utils/src/error.ts:67]: @alexvy86 @anthony-murphy Any thoughts on the naming and documentation here? I used "MessageLike" as a temporary placeholder.

Comment (anthony-murphy) [reply packages/utils/telemetry-utils/src/error.ts:67]: i would prefer it we didn't create an explicity type, and it only existed in the function, as that is the only place it makes sense

Comment (anthony-murphy) [reply packages/utils/telemetry-utils/src/error.ts:67]: also, i would mark everything here internal

Comment (anthony-murphy) [reply packages/utils/telemetry-utils/src/error.ts:67]: really everything in this package should be internal.

Comment (Josmithr) [reply packages/utils/telemetry-utils/src/error.ts:67]: Okay. I can re-inline it if it isn't something that stands on its own. I'd still be interested in documenting the type semantics in the two places that use it. Any recommendations for that?

And when you say "mark everything here internal", do you mean these 2 moved items, or **all** of the contents of this module?

Comment (Josmithr) [reply packages/utils/telemetry-utils/src/error.ts:67]: > really everything in this package should be internal.

Should this package move to the _new_ "@fluid-internal" scope then?

File: packages/utils/telemetry-utils/src/error.ts [hunk 2]
```diff
@@ -53,3 +60,118 @@ export class DataCorruptionError extends LoggingError implements IErrorBase, IFl
...
+			dataProcessingError.addTelemetryProperties(normalizedError.getTelemetryProperties());
+
+			return dataProcessingError;
+		}
+		return normalizedError;
+	}
+}
+
+/**
+ * Extracts specific properties from the incoming message that we know are safe to log.
```
[the remark this answers is not in the document]
Comment (anthony-murphy) [reply packages/utils/telemetry-utils/src/error.ts:172]: it is also used in the pending state manager to capture parts of an outgoing message

File: packages/utils/telemetry-utils/src/error.ts (hunk 1 above)
Comment (anthony-murphy) [reply packages/utils/telemetry-utils/src/error.ts]: that seems wrong, we still need to use it in our public packages, we just don't want to support external usage of the surface area.

Comment (Josmithr) [reply packages/utils/telemetry-utils/src/error.ts]: Tyler is redefining that scope (or creating a new one, I forget where that landed). What I'm referring to is a scope that is published publicly, but documented as not being suitable for external consumption. And that seems to be the intent for this package based on Tyler's spreadsheet.

File: packages/utils/telemetry-utils/src/error.ts [hunk 3]
```diff
@@ -47,9 +54,138 @@ export class UsageError extends LoggingError implements IUsageError, IFluidError
  */
 export class DataCorruptionError extends LoggingError implements IErrorBase, IFluidErrorBase {
 	readonly errorType = FluidErrorTypes.dataCorruptionError;
-	readonly canRetry = false;
+	// readonly canRetry = false;
```
Comment (anthony-murphy) [inline packages/utils/telemetry-utils/src/error.ts:57]: why this change?

**Review by anthony-murphy — APPROVED** [review]
couple small comments. otherwise looks good.

File: packages/utils/telemetry-utils/src/error.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexvy86) [reply packages/utils/telemetry-utils/src/error.ts]: Nothing else to add from my side, not really that familiar with the function.

**Review by alexvy86 — APPROVED** [review]
Just Tony's comment about that line which looks like it was commented out by mistake, otherwise lgtm.

File: packages/utils/telemetry-utils/src/error.ts (hunk 3 above)
Comment (Josmithr) [reply packages/utils/telemetry-utils/src/error.ts:57]: I hadn't intended to push that. Good catch - thanks!

Comment (Josmithr) [reply packages/utils/telemetry-utils/src/error.ts:57]:
```suggestion
	readonly canRetry = false;
```
